Riot/Modules/Authentication/AuthenticationCoordinator.swift

```diff
@@ -130,9 +130,19 @@ final class AuthenticationCoordinator: NSObject, AuthenticationCoordinatorProtoc
         }
 
         let flow: AuthenticationFlow = initialScreen == .login ? .login : .register
+
+        // Use the homeserver defined by a provisioningLink or by the user (if none is set, the default one will be used)
+        let homeserverAddress = authenticationService.provisioningLink?.homeserverUrl ?? authenticationService.state.homeserver.addressFromUser
+        
+        // Check if the user must select a server
+        if BuildSettings.forceHomeserverSelection, homeserverAddress == nil {
```

Comment:
First off, good job remembering to check the provisioning link - I would have forgotten about that 😃

However we shouldn't include the `addressFromUser` here as it alters the intended behaviour where navigating back to the carousel and starting again defaults to matrix.org once more. Could probably remove the `let homeserverAddress` here, check the provisioning link directly in the if and then call `startFlow(flow)` like before without an address.

Reply:
Perfect, thanks 👍